src/export/StructureDefinitionExporter.ts

```diff
@@ -414,14 +414,38 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        if (element.path !== element.base.path) {
-          // The AddElementRule always sets the newElementBase.path to the value of element.path.
-          // All parent elements will have the element.base.path pointing to the parent.
-          logger.error(
-            `FHIR prohibits logical models and resources from constraining parent elements. Skipping '${rule.constructorName}' at path '${rule.path}' for '${fshDefinition.name}'.`,
-            rule.sourceInfo
-          );
-          continue;
+        const baseElementPath = element.path.replace(element.structDef.id, '').split('.')[1];
```

Comment:
I'm not sure that just taking the element at index 1 is correct. Is it possible for the element we're looking at to be more deeply nested, such that its parent path would include more parts?

Reply:
That's certainly possible. My thinking here was that for elements other than `BackboneElement`s, it would actually be preferable to check for the top-level element rather than the entirety of the path. That was based on an assumption that you couldn't use an `addElementRule` to add nested elements to a pre-defined fhir type, e.g `HumanName.foo`. 

When I just checked on that though, I wasn't able to find anything in the spec that rules that out. It looks like the IGPublisher doesn't have a problem with it either, so I'll rework this to be a bit more granular. What I was trying to avoid was a situation where a parent `Resource`/`Logical` adds an element of type `HumanName` for example, then a child `Resource`/`Logical` constrains a sub-element of `HumanName` and that's allowed because the path _looks_ like a new element even though it's technically inherited. I'll address this now!

Reply:
Sorry... just seeing this now, but adding new subelements to concrete types (like `HumanName`) is **illegal**, even if IG Publisher does not complain about it.  I asked this in Zulip a few weeks back: https://chat.fhir.org/#narrow/stream/179177-conformance/topic/Logical.20Models.3A.20Adding.20new.20sub-elements.20to.20existing.20types.3F

So, it might be ok to add a new subelement to an element w/ type `Element` or `BackboneElement`, but I think that's it.

Reply:
I figured this was the case! I opted to just determine if a rule is legal by checking if an `addElementRule` is present with a matching path. That seemed like a pretty surefire way to know if the element was inherited or not.